fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/NormalizedLuceneSearchProvider.java

```diff
@@ -0,0 +1,163 @@
+                if (lastIndexSearcher != luceneIndexSearcher) {
+                    lastIndexSearcher = luceneIndexSearcher;
+                    searchColumnExists = validateSearchColumn(SEARCH_COLUMN_NAME);
+                }
+            }
+        }
+        if (!searchColumnExists) {
+            throw new UnsupportedOperationException(
+                    String.format(
+                            "Underlying LuceneIndex for Dimension %s does not support search queries.",
+                            getDimension()
+                    )
+            );
+        }
+
+        return getResultsPage(getSearchQuery(searchQueryString), paginationParameters);
+    }
+
+    @Override
+    public void setDimension(Dimension dimension) {
+        super.setDimension(dimension);
+
+        analyzer = buildNewAnalyzerMapFromDimension(getDimension());
+        queryParser = new SimpleQueryParser(analyzer, SEARCH_COLUMN_NAME);
+        queryParser.setDefaultOperator(BooleanClause.Occur.MUST);
+    }
+
+    /**
+     * Builds a new analyzer map using the provided dimension. Maps all fields of the provided dimension to use the
+     * standard lucene analyzer and sets the search column to use the diacritic unaware analyzer.
+     *
+     * @param dimension Dimension to build the new analyzer map off of
+     * @return the new analyzer
+     */
+    protected Analyzer buildNewAnalyzerMapFromDimension(Dimension dimension) {
+        Map<String, Analyzer> analyzerMap = dimension.getDimensionFields().stream()
+                .map(field -> DimensionStoreKeyUtils.getColumnKey(field.getName()))
+                .collect(Collectors.toMap(
+                        Function.identity(),
+                        unused -> STANDARD_LUCENE_ANALYZER
+                ));
+        analyzerMap.put(SEARCH_COLUMN_NAME, DIACRITIC_ANALYZER);
+
+        return new PerFieldAnalyzerWrapper(STANDARD_LUCENE_ANALYZER, analyzerMap);
+    }
+
+    /**
+     * Generate the Lucene search query from the provided search query string. The search query string is intended to
+     * be provided by a user.
+     *
+     * @param searchQueryString  the query string to convert into a Lucene Query object
+     * @return the query object
+     */
+    private Query getSearchQuery(String searchQueryString) {
+        return queryParser.parse(QueryParser.escape(searchQueryString));
+    }
+
+    /**
+     * Validates that the search column with the provided name exists, by checking that more than 0 documents have a
+     * field with the provided name.
+     *
+     * @param searchColumnName The name of the search column to check.
+     * @return whether or not that field exists in the lucene index.
+     */
+    protected boolean validateSearchColumn(String searchColumnName) {
+        initializeIndexSearcher();
+        readLock();
+        try {
+            int docCount = luceneIndexSearcher.getIndexReader().getDocCount(searchColumnName);
+            return (docCount > 0);
```

Comment:
The parenthesis are not necessary. Also, should be able to just do `luceneIndexSearcher.getIndexReader().getDocCount(searchColumnName) > 0`